pkg/handler/handler.go

```diff
@@ -34,32 +34,32 @@ var log = logf.Log.WithName("handler")
 func WebhookHandler(w http.ResponseWriter, r *http.Request, reconciler gitopsconfig.Reconciler) {
 	log.Info("received webhook call")
 	if r.Method != "POST" {
+		log.Info("webhook handler only accepts the POST method", "sent_method", r.Method)
 		w.WriteHeader(405)
 		return
 	}
-	//log.Info("webhook is of type post")
+
 	payload, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		log.Error(err, "error reading request body")
 		return
 	}
 	defer r.Body.Close()
-	//log.Info("parsed body")
+
 	event, err := github.ParseWebHook(github.WebHookType(r), payload)
 	if err != nil {
-		log.Error(err, "could not parse webhook")
+		log.Error(err, "error parsing webhook event payload")
 		return
 	}
-	//log.Info("parsed body, found event", "event", event)
+
 	switch e := event.(type) {
 	case *github.PushEvent:
-		// this is a commit push, do something with it
+		// A commit push was received, determine if there is are GitOpsConfigs that match the event
+		// The repository url and Git ref must match for the templateSource or parameterSource
 		{
-			//find the list of CR that have this url.
-			//log.Info("event is of type push")
 			list, err := reconciler.GetAll()
 			if err != nil {
-				log.Error(err, "unable to get the list of GitOpsCionfig")
+				log.Error(err, "error getting the list of GitOpsConfigs")
```

Comment:
It would be helpful to state that this is within the Webhook Handler. Same goes for all error messages.

Reply:
@Smiley73 the error stacktrace provides the package, function, and line number of the error.
We could add it if it would make it more readable, but the information is already available.
Example:
```
{
    "level": "error",
    "ts": 1591223803.52741,
    "logger": "handler",
    "msg": "error parsing webhook event payload",
    "error": "unknown X-Github-Event in message: ",
    "stacktrace": "github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/KohlsTechnology/eunomia/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/KohlsTechnology/eunomia/pkg/handler.WebhookHandler\n\t/go/src/github.com/KohlsTechnology/eunomia/pkg/handler/handler.go:51\nmain.main.func1\n\t/go/src/github.com/KohlsTechnology/eunomia/cmd/manager/main.go:183\nnet/http.HandlerFunc.ServeHTTP\n\t/go/src/net/http/server.go:2007\nnet/http.(*ServeMux).ServeHTTP\n\t/go/src/net/http/server.go:2387\nnet/http.serverHandler.ServeHTTP\n\t/go/src/net/http/server.go:2802\nnet/http.(*conn).serve\n\t/go/src/net/http/server.go:1890"
}
```

Reply:
It might help our user base to have a clearer error message, without having to dig through the stack trace. I can go either way on this one.

Reply:
The log formatting for the entire operator is controlled here ... https://github.com/KohlsTechnology/eunomia/blob/master/cmd/manager/main.go#L77-L85. There is probably a way to configure the log format to add the filename and line number. This will be simpler than adding this information to every single log statement.

Here are the docs we need to read https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.3.0/pkg/log?tab=doc

Reply:
@seanmalloy / @Smiley73 I updated the 4 loggers to include the filename as additional labels. I was trying to figure out how to configure this at the top level of the logger configuration but could not get it to work. Looking at [examples in the controller-runtime repo](https://github.com/kubernetes-sigs/controller-runtime/tree/master/examples), I didn't see any additional configuration being added besides adding labels as I have done in c5976dd .

It looks like we could configure this in zap, but I could not get this to work with our code. I was utilizing this [basic configuration example](https://pkg.go.dev/go.uber.org/zap@v1.10.0?tab=doc#example-package-BasicConfiguration) in the documentation for zap and [this zap issue](https://github.com/uber-go/zap/issues/618#issuecomment-411680918) which looks like it would include line number and filename.